posthog/cdp/templates/attio/template_attio.py

```diff
@@ -1,6 +1,259 @@
+    status="beta",
+    free=False,
+    type="destination",
+    id="template-attio-workspace",
+    name="Attio",
+    description="Create and update Workspace and Company records in Attio",
+    icon_url="/static/services/attio.png",
+    category=["CRM", "Customer Success"],
+    code_language="hog",
+    code="""
+let companyBody := {
+    'data': {
+        'values': {
+            'domains': [
+                {
+                    'domain': inputs.companyDomain
+                }
+            ]
+        }
+    }
+}
+
+let workspaceBody := {
+    'data': {
+        'values': {
+            'workspace_id': inputs.workspaceId,
+            'company': inputs.companyDomain,
+        }
+    }
+}
+
+if (not empty(inputs.userId)) {
+    workspaceBody.data.values.users := [inputs.userId]
+}
+
+for (let key, value in inputs.workspaceAttributes) {
+    if (not empty(value)) {
+        workspaceBody.data.values[key] := value
+    }
+}
+
+for (let key, value in inputs.companyAttributes) {
+    if (not empty(value)) {
+        companyBody.data.values[key] := value
+    }
+}
+
+let companyRes := fetch(f'https://api.attio.com/v2/objects/companies/records?matching_attribute=domains', {
+    'method': 'PUT',
+    'headers': {
+        'Authorization': f'Bearer {inputs.apiKey}',
+        'Content-Type': 'application/json',
+    },
+    'body': companyBody
+})
+if (companyRes.status >= 400) {
+    throw Error(f'Error creating Company from api.attio.com (status {companyRes.status}): {companyRes.body}')
+}
+
+let workspaceRes := fetch(f'https://api.attio.com/v2/objects/workspaces/records?matching_attribute=workspace_id', {
+    'method': 'PUT',
+    'headers': {
+        'Authorization': f'Bearer {inputs.apiKey}',
+        'Content-Type': 'application/json',
+    },
+    'body': workspaceBody
+})
+if (workspaceRes.status >= 400) {
+    throw Error(f'Error creating Workspace from api.attio.com (status {workspaceRes.status}): {workspaceRes.body}')
+}
+""".strip(),
+    inputs_schema=[
+        {
+            "key": "apiKey",
+            "type": "string",
+            "label": "Access token",
+            "description": "Check out this page to get your API key: https://attio.com/help/reference/integrations-automations/generating-an-api-key",
+            "secret": True,
+            "required": True,
+        },
+        {
+            "key": "userId",
+            "type": "string",
+            "label": "User ID",
+            "description": "ID of the user to link to this workspace (optional). Leave blank to skip user linking.",
+            "default": "{person.uuid}",
+            "secret": False,
+            "required": False,
+        },
+        {
+            "key": "workspaceId",
+            "type": "string",
+            "label": "Workspace ID",
+            "description": "ID of this workspace in your database. Use to reference the relevant Workspace record in Attio",
+            "default": "{event.properties.$group_key}",
```

Comment:
```suggestion
            "default": "{event.properties.$group_0}",
```

> The one question I have is: is it safe to use $group_key as a stable ID representing a workspace? This seemed correct in my testing, but I saw that Accoil have a more in-depth handling of group IDs in [their own Posthog integration](https://developer.accoil.com/docs/posthog).

A posthog event can hold up to five different group properties, `$group_key` will specify the key for which `posthog.group` has been called. Ideally, you'll set this to the group that represents the workspace, which is different for every user. I'd default this value to `event.properties.$group_0`